neqo-transport/src/recovery.rs

```diff
@@ -581,13 +588,14 @@ impl PtoState {
     }
 
     /// Generate a sending profile, indicating what space it should be from.
-    /// This takes a packet from the supply or returns an ack-only profile if it can't.
-    pub fn send_profile(&mut self, mtu: usize) -> SendProfile {
+    /// This takes a packet from the supply if one remains, or returns `None`.
+    pub fn send_profile(&mut self, mtu: usize) -> Option<SendProfile> {
         if self.packets > 0 {
+            // This is a PTO, so ignore the limit.
             self.packets -= 1;
-            SendProfile::new_pto(self.space, mtu, self.probe)
+            Some(SendProfile::new_pto(self.space, mtu, self.probe))
         } else {
-            SendProfile::new_limited(0)
+            None
```

Comment:
Note that this now returns `None` rather than returning a profile that only permits ACK frames.

This allows us to send probes on a path between the first and second PTO.  I found that we would not send anything at that time because - although we had plenty of congestion control window available - we were sending on PTO and therefore blocked.

This change might be a help if we have new data to send when the PTO timer is otherwise active.  This might be the source of some of our stalls.